lib/Doctrine/ORM/Cache/Persister/Entity/CachedEntityPersister.php

```diff
@@ -44,4 +44,9 @@ public function getEntityHydrator();
      * @return boolean
      */
     public function storeEntityCache($entity, EntityCacheKey $key);
+
+    /**
+     * @return \Doctrine\ORM\Cache\TimestampCacheKey
+     */
+    public function getTimestampKey();
```

Comment:
Also, I don't know how to regenerate a timestamp key, if not retrieved from here.